lib/api/link_builder.rb

```diff
@@ -0,0 +1,81 @@
+module Api
+  class LinkBuilder
+    PAGING_LINKS = [
+      :self,
+      :next,
+      :previous,
+      :first,
+      :last
+    ].freeze
+
+    attr_reader :offset, :limit, :href, :count, :subcount, :subquery_count
+
+    def initialize(params, href, counts)
+      @offset = params["offset"].to_i if params["offset"]
```

Comment:
@imtayadeway not a bad suggestion, the only reason why I am not entirely sold on it is the rebuilding part is a bit simpler than the parsing part, and if we already have those parameters, I would prefer to just use those rather than re-parsing it. If that makes sense? Open to discuss though.